lib/modules/versioning/hashicorp/index.spec.ts

```diff
@@ -13,9 +14,10 @@ describe('modules/versioning/hashicorp/index', () => {
   );
 
   it.each`
-    versions                                         | range         | expected
-    ${['0.4.0', '0.5.0', '4.0.0', '4.2.0', '5.0.0']} | ${'~> 4.0'}   | ${'4.2.0'}
-    ${['0.4.0', '0.5.0', '4.0.0', '4.2.0', '5.0.0']} | ${'~> 4.0.0'} | ${'4.0.0'}
+    versions                                         | range                 | expected
+    ${['0.4.0', '0.5.0', '4.0.0', '4.2.0', '5.0.0']} | ${'~> 4.0'}           | ${'4.2.0'}
+    ${['0.4.0', '0.5.0', '4.0.0', '4.2.0', '5.0.0']} | ${'~> 4.0.0'}         | ${'4.0.0'}
+    ${['0.4.0', '0.5.0', '4.0.0', '4.2.0', '5.0.0']} | ${'!=4.2.0, > 4.0.0'} | ${'5.0.0'}
```

Comment:
```suggestion
    ${['0.4.0', '0.5.0', '4.1.0', '4.2.0', '5.0.0']} | ${'!=4.2.0, > 4.0.0'} | ${'4.1.0'}
```

Reply:
getSatisfyingVersion returns highest among the supplied array, so it will be `5.0.0` right?

Reply:
Ah, I have misread that. 

Add my edit as additional test